### DISCUSSION TITLE
Printing diffs

### DISCUSSION THREAD

**Discussion by mistotebe** [Textualize/rich#1610] [Q&A]
Hi, I'm processing some logs and want to highlight changes to state (tracked by e.g. a dictionary) but can't find a way to do so. What I was thinking of doing:

When state changes from `{'removed': 'value', 'kept': 'something'}` to `{'kept': 'completely different', 'added': 'new'}` I'd like to display the following, but still have it behave as a dictionary (or set, or something) when pretty printing:
```
{'[red]removed[/]': '[red]value[/]', 'kept': '[yellow]completely different[/]', '[green]added[/]': '[green]new[/]'}
```

It is trivial to generate the appropriate data, but I have no idea how to pass it to rich for printing (neither `__rich__()` nor `__rich_console__()` seem to give me the right interface).

**Comment by willmcgugan** [comment]
Can't think of any shortcuts. Would you want it to be recursive? Might be complicated. Have a look at the Pretty class, which may be a good place to start.

**Reply by mistotebe** [reply]
I was thinking of any `str()` object being printed (regardless where it came from) could be formatted as usual, still haven't figured out how to achieve that...
